lasagne/layers/base.py

```diff
@@ -23,8 +23,10 @@ class Layer(object):
     network's output :class:`Layer` instance doubles as a handle to the full
     network.
     """
-    def __init__(self, input_layer):
-        self.input_layer = input_layer
+    def __init__(self, input_layer, name=None):
+        if input_layer is not None:
+            self.input_layer = input_layer
```

Comment:
No, currently the presence of `self.input_layer` or `self.input_layers` is used by `lasagne.layers.get_all_layers` to traverse the graph: https://github.com/benanne/Lasagne/blob/master/lasagne/layers/helper.py#L57
So `InputLayer` is not supposed to have it.

Of course we could make it so this property exists for `InputLayer`, but is set to `None`. Then we just have to update `get_all_layers` to check for `None`. Either approach is fine for me.